com.unity.multiplayer.mlapi/Tests/Runtime/NetworkShowHideTests.cs

```diff
@@ -0,0 +1,162 @@
+using UnityEngine;
+using UnityEngine.TestTools;
+using Unity.Netcode;
+
+public class NetworkShowHideTest : NetworkBehaviour
+{
+
+}
+
+public class ShowHideObject : NetworkBehaviour
+{
+
+}
+
+namespace Unity.Netcode.RuntimeTests
+{
+    public class NetworkShowHideTests : BaseMultiInstanceTest
+    {
+        protected override int NbClients => 2;
+
+        private ulong m_ClientId0;
+
+        private NetworkObject m_NetSpawnedObject1;
+        private NetworkObject m_NetSpawnedObject2;
+        private NetworkObject m_NetSpawnedObject3;
+        private NetworkObject m_Object1OnClient0;
+        private NetworkObject m_Object2OnClient0;
+        private NetworkObject m_Object3OnClient0;
+
+        [UnitySetUp]
+        public override IEnumerator Setup()
+        {
+            yield return StartSomeClientsAndServerWithPlayers(useHost: true, nbClients: NbClients,
+                updatePlayerPrefab: playerPrefab =>
+                {
+                    var networkTransform = playerPrefab.AddComponent<NetworkShowHideTest>();
+                });
+        }
+
+        public GameObject PreparePrefab(Type type)
+        {
+            var prefabToSpawn = new GameObject();
+            prefabToSpawn.AddComponent(type);
+            var networkObjectPrefab = prefabToSpawn.AddComponent<NetworkObject>();
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(networkObjectPrefab);
+            m_ServerNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab() { Prefab = prefabToSpawn });
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                clientNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab() { Prefab = prefabToSpawn });
+            }
+            return prefabToSpawn;
+        }
+
+        // Check that the first client see them, or not, as expected
+        private void checkVisible(bool target)
+        {
+            Debug.Assert(m_NetSpawnedObject1.IsNetworkVisibleTo(m_ClientId0) == target);
+            Debug.Assert(m_NetSpawnedObject2.IsNetworkVisibleTo(m_ClientId0) == target);
+            Debug.Assert(m_NetSpawnedObject3.IsNetworkVisibleTo(m_ClientId0) == target);
+
+            Debug.Assert(m_Object1OnClient0.IsSpawned == target);
+            Debug.Assert(m_Object2OnClient0.IsSpawned == target);
+            Debug.Assert(m_Object3OnClient0.IsSpawned == target);
+        }
+
+        // Set the 3 objects visibility
+        private void show(bool visibility)
+        {
+            var list = new List<NetworkObject>();
+            list.Add(m_NetSpawnedObject1);
+            list.Add(m_NetSpawnedObject2);
+            list.Add(m_NetSpawnedObject3);
+
+            if (!visibility)
+            {
+                NetworkObject.NetworkHide(list, m_ClientId0);
+            }
+            else
+            {
+                NetworkObject.NetworkShow(list, m_ClientId0);
+            }
+        }
+
+        private IEnumerator refreshNetworkObjects()
```

Comment:
huh, interesting. camelCase private method names are not giving us naming errors, weird :P

Reply:
Will fix, thanks. But see what you did with your `standards.py` ? Now I rely blindly on it to fix my casing and forgot to think in the first place 😄

Reply:
wait, doesn't `standards.py` gives you an error in this case? maybe we forgot to have that rule in the ruleset. I need to check Unity coding standards and our `standards.py` implementation.